### PR TITLE
Create database operation between client > server > MongoDB

### PR DESCRIPTION
**Pull request by hauptdigital**
Deployment: https://dev.deepspeech-notes.haupt.digital/

In this pull request I setup this functionality:

- When you see an empty note and enter something in the title or content, it will create a new note in the database with your first `onBlur` (clicking on something different or focus out of the editable areas).
- When a note is accessed with a valid note Id in the URL parameters, it will display the title and content of the note
- When you record a note and hit "stop recording", it will save the current note.

For this functionality, developed these api function:

- `getNote` ->  (GET request) Uses the id from the URL parameter to get title and content of a note.
- `postNote` -> (POST request) Creates a new note and return the generated note id.
- `updateNote` -> (PATCH request) Edits an existing note using a valid note Id (either from URL Parameter or set by initial creation of a note).
- `getNotes` ->  (GET request) Get all notes (currently not in use but I thought it makes sense to develop this now).

mongodb

### CODE REVIEW

File: client/src/api/notes.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+export function getNote(noteId) {
+  return fetch(`/api/notes/${noteId}`, {
+    method: 'GET',
+  })
+    .then((response) => {
+      if (response.status !== 200) {
```
Comment (lmachens) [inline client/src/api/notes.js:6]: Are you sure you want to throw an error for the other cases like `301`?
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
Check **Checking that the fetch was successful**

```suggestion
      if (!response.ok) {
```

(happens multiple times)

File: client/src/api/notes.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    })
+    .then((response) => response.json());
+}
+
+export function postNote(note) {
+  return fetch(`/api/notes/`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Encrypt: 'true',
```
Comment (lmachens) [inline client/src/api/notes.js:32]: Looks like a copy-paste from copypasta ;)

```suggestion
```

Same in your PATCH method

File: client/src/pages/Notes.js [hunk 3]
```diff
@@ -41,18 +58,34 @@ function Notes() {
...
   }
 
+  React.useEffect(() => {
+    // Get note title and content if noteId is set
+    async function doGetNote(noteId) {
+      const note = await getNote(noteId);
+      return note;
+    }
+    if (noteId) {
+      doGetNote(noteId).then((note) => {
```
Comment (lmachens) [inline client/src/pages/Notes.js:75]:
```suggestion
    if (noteId) {
      getNote(noteId).then((note) => {
```

File: client/src/pages/Notes.js [hunk 4]
```diff
@@ -70,18 +103,20 @@ function Notes() {
...
           onChange={handleNoteTitleChange}
+          onBlur={saveNote}
           value={noteTitle}
           placeholder={placeholders.title}
         />
         {isRecording ? (
           <NoteContentReadOnly
             noteContent={noteContent}
-            placeholder={placeholders.note}
+            placehonplder={placeholders.note}
```
Comment (lmachens) [inline client/src/pages/Notes.js:113]:
```suggestion
            placeholder={placeholders.note}
```

File: client/src/api/notes.js (hunk 2 above)
Comment (hauptdigital) [reply client/src/api/notes.js:32]: yikes :-)

**Pull request merged**